src/main/java/org/radarbase/hdfs/Application.kt

```diff
@@ -0,0 +1,258 @@
+                    logger.info("Out: {}", pathFactory.root)
+                    restructure.process(input)
+                }
+
+                val numberFormat = NumberFormat.getNumberInstance()
+                logger.info("Processed {} files and {} records in {}",
+                        numberFormat.format(restructure.processedFileCount),
+                        numberFormat.format(restructure.processedRecordsCount),
+                        timeStart.durationSince().formatTime())
+            }
+        } catch (ex: IOException) {
+            logger.error("Processing failed", ex)
+        } catch (e: InterruptedException) {
+            logger.error("Processing interrupted")
+        }
+    }
+
+    private fun Temporal.durationSince() = Duration.between(this, Instant.now())
+
+    class Builder(val settings: RestructureSettings, val hdfsSettings: HdfsSettings) {
+        var storageDriver: StorageDriver? = null
+        var pathFactory: RecordPathFactory? = null
+        var compressionFactory: CompressionFactory? = null
+        var formatFactory: FormatFactory? = null
+        val properties = HashMap<String, String>()
+        var inputPaths: List<String> = listOf()
+        var asService: Boolean = false
+        var pollInterval: Int = 0
+        var lockPath: Path? = null
+
+        @Throws(IOException::class)
+        fun build(): Application {
+            pathFactory = (pathFactory ?: ObservationKeyPathFactory())
+                    .also { it.init(properties) }
+            storageDriver = (storageDriver ?: LocalStorageDriver())
+                    .also { it.init(properties) }
+            compressionFactory = (compressionFactory ?: CompressionFactory())
+                    .also { it.init(properties) }
+            formatFactory = (formatFactory ?: FormatFactory())
+                    .also { it.init(properties) }
+
+            return Application(this)
+        }
+    }
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(Application::class.java)
+        const val CACHE_SIZE_DEFAULT = 100
+
+        @JvmStatic
+        fun main(args: Array<String>) {
+            val commandLineArgs = CommandLineArgs()
+            val parser = JCommander.newBuilder().addObject(commandLineArgs).build()
+
+            parser.programName = "radar-hdfs-restructure"
+            try {
+                parser.parse(*args)
+            } catch (ex: ParameterException) {
+                logger.error(ex.message)
+                parser.usage()
+                exitProcess(1)
+            }
+
+            if (commandLineArgs.help) {
+                parser.usage()
+                exitProcess(0)
+            }
+
+            logger.info(SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(Date()))
+            logger.info("Starting...")
+
+            Timer.isEnabled = commandLineArgs.enableTimer
+
+            if (commandLineArgs.enableTimer) {
+                Runtime.getRuntime().addShutdownHook(Thread(
```

Comment:
I don't the HDFS-restrucute much. I am just trying to understand why do we run this timer ina  shutDownHook when one is trying to enableTimer?